src/System.Slices/src/System/PtrUtils.cs

```diff
@@ -32,43 +32,50 @@ sealed class PtrUtils
         // depends on it working... (okay, I still feel a little dirty.)
 
         /// <summary>
-        /// Takes a (possibly null) object reference, plus an offset in bytes,
+        /// Takes a (possibly null) object reference, plus an offset in bytes, plus an index,
         /// adds them, and safetly dereferences the target (untyped!) address in
         /// a way that the GC will be okay with.  It yields a value of type T.
         /// </summary>
-
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [ILSub(@"
-            .maxstack 2
-            .locals ([0] uint8& addr)
+        [ILSub(@"            
+            .maxstack 3
+            .locals([0] uint8 & addr)
             ldarg.0     // load the object
             stloc.0     // convert the object pointer to a byref
             ldloc.0     // load the object pointer as a byref
             ldarg.1     // load the offset
             add         // add the offset
+            ldarg.2     // load the index
+            sizeof !!T  // load size of T
+            conv.u  
```

Comment:
Indeed, `native int x int32` is valid. It's `native int x int64` that isn't allowed.